test/data/decode/decoding.js

```diff
@@ -0,0 +1,173 @@
+import debugModule from "debug";
+const debug = debugModule("test:data:decode");
+
+import { assert } from "chai";
+import Ganache from "ganache-cli";
+import Web3 from "web3";
+import util from "util";
+
+import { prepareContracts } from "test/helpers";
+
+import Debugger from "lib/debugger";
+
+import { cleanBigNumbers } from "lib/data/decode/utils";
+
+import data from "lib/data/selectors";
+import evm from "lib/evm/selectors";
+
+const __STORAGE = `pragma solidity ^0.4.18;
+
+contract StorageVars {
+  event Ran();
+
+  string shortString;
+  string longString;
+
+  uint[] multipleFullWordArray;  // length 3, will take up 3 words
+  uint16[] withinWordArray;  // length 10, will take up >1/2 word
+  uint64[] multiplePartWordArray; // length 9, will take up 2.25 words
+
+  uint240[] inconvenientlyWordOffsetArray; // length 3, takes ~2.8 words
+
+  function run() public {
+    uint start = 1;
+    uint i;
+
+    shortString = "hello world";
+    longString = "solidity storage is a fun lesson in endianness";
+
+    for (i = 0; i < 3; i++) {
+      multipleFullWordArray.push(start + i);
+    }
+    start = i;
+
+    for (i = 0; i < 10; i++) {
+      withinWordArray.push(uint16(start + i));
+    }
+    start = i;
+
+    for (i = 0; i < 9; i++) {
+      multiplePartWordArray.push(uint64(start + i));
+    }
+    start = i;
+
+    for (i = 0; i < 3; i++) {
+      inconvenientlyWordOffsetArray.push(uint240(start + i));
+    }
+    start = i;
+
+    Ran();
+  }
+}
+`;
+
+function imitateRun() {
+  let multipleFullWordArray = [];
+  let withinWordArray = [];
+  let multiplePartWordArray = [];
+  let inconvenientlyWordOffsetArray = [];
+  let shortString = "hello world";
+  let longString = "solidity storage is a fun lesson in endianness";
+
```

Comment:
Suspect `longString === false` here, not sure.

Reply:
You know, `fun` in the dwarf-fortress sense of the word http://dwarffortresswiki.org/index.php/Fun

Reply:
LOL!